server/test/views/questiontypes/DateQuestionRendererTest.java

```diff
@@ -0,0 +1,110 @@
+package views.questiontypes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.test.Helpers.stubMessagesApi;
+
+import com.google.common.collect.ImmutableSet;
+import j2html.tags.specialized.DivTag;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.OptionalLong;
+import org.junit.Before;
+import org.junit.Test;
+import play.i18n.Lang;
+import play.i18n.Messages;
+import services.LocalizedStrings;
+import services.applicant.ApplicantData;
+import services.applicant.question.ApplicantQuestion;
+import services.question.types.DateQuestionDefinition;
+import services.question.types.QuestionDefinitionConfig;
+import views.questiontypes.ApplicantQuestionRendererParams.ErrorDisplayMode;
+
+public class DateQuestionRendererTest {
```

Comment:
I added test files for question types that didn't have them already.

Reply:
Thanks for adding these tests!  Yes, there's some redundancy, but I agree that each question type should have it's own test in case of any differences.